src/EcommerceStatsBundle/Controllers/CsvExportController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\EcommerceStatsBundle\Controllers;
+
+use ChameleonSystem\EcommerceStatsBundle\DataModel\CsvResponse;
+use ChameleonSystem\EcommerceStatsBundle\Interfaces\CsvExportServiceInterface;
+use ChameleonSystem\EcommerceStatsBundle\Interfaces\StatsTableServiceInterface;
+use ChameleonSystem\EcommerceStatsBundle\Interfaces\TopSellerServiceInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\VarDumper\VarDumper;
+
+class CsvExportController
+{
+
+    /**
+     * @var CsvExportServiceInterface
+     */
+    private $csvExportService;
+
+    /**
+     * @var StatsTableServiceInterface
+     */
+    private $statsTableService;
+
+    /**
+     * @var TopSellerServiceInterface
+     */
+    private $topSellerService;
+
+    public function __construct(
+        CsvExportServiceInterface $csvExportService,
+        StatsTableServiceInterface $statsTableService,
+        TopSellerServiceInterface $topSellerService
+    ) {
+        $this->csvExportService = $csvExportService;
+        $this->statsTableService = $statsTableService;
+        $this->topSellerService = $topSellerService;
+    }
+
+    public function exportStatistics(Request $request): Response
```

Comment:
Request should be a (service) dependency.

Reply:
I don't really see a difference between having symfony inject a `RequestStack` in the constructor and having symfony inject a `Request` in the controller action. In the contrary: I would argue that the request is unique to each action and having it injected in each action makes that a bit clearer IMO.